File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = l + 1
        maxProfit = 0
        while(l < r and r < len(prices)):
            maxProfit = max(maxProfit, prices[r]-prices[l])
            if prices[r] < prices[l]:
                l = r
                r += 1
            else:
                r += 1
        return maxProfit

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l =0, n=s.size(), maxL=0;
        unordered_set<int> uset{};
        for(int r=0; r<n; r++){
            while(uset.count(s[r])){
                uset.erase(s[l]);
                l++;
            }
            uset.insert(s[r]);
            maxL = max(maxL, r-l+1);
        }
        return maxL;
    }
};
